### ISSUE TITLE
Migration to 0.15.1 takes very long

### ISSUE THREAD

**Issue by Deekee62** [lightningnetwork/lnd#6890]
### Background
My channel.db is quite big 20 GB, so I decided  to upgrade from v0.14.2-beta to v0.15.1-beta, but it seems that the migration takes forever!

3-SEP-22  - Started the migration:
`2022-09-03 14:06:07.940 [INF] CHDB: Performing database optional migration: prune revocation log
2022-09-03 14:06:25.874 [INF] CHDB: Migration progress: 0.049%, still have: 8336642`

In the beginning it was quite fast:
`2022-09-03 15:38:12.907 [INF] CHDB: Migration progress: 40.615%, still have: 4953129`

4-SEP-22  - But then it started to slow down drastically:
`2022-09-04 00:10:30.349 [INF] CHDB: Migration progress: 54.548%, still have: 3791032`
`2022-09-05 00:09:05.290 [INF] CHDB: Migration progress: 63.412%, still have: 3051668`

5-SEP-22 - The last change was this morning, since then no progress:
`2022-09-05 04:34:19.464 [INF] CHDB: Migration progress: 71.766%, still have: 2354916`
`2022-09-05 12:06:03.455 [INF] CHDB: Migration progress: 71.766%, still have: 2354916`

While writing this issue I had some little progress:-)
`2022-09-05 12:28:11.981 [INF] CHDB: Migration progress: 77.409%, still have: 1884183`

### Your environment
LND Version: 0.15.1-beta
Linux umbrel 5.10.17-v8+ 1403 SMP PREEMPT Mon Feb 22 11:37:54 GMT 2021 aarch64 GNU/Linux
Bitcoin Version: 220000,
Subversion: Satoshi:22.0.0
Protocol Version: 70016
Hardware Details
Hardware: BCM2835
Revision: d03114
Serial: 100000000b2d1319
Model: Raspberry Pi 4 Model B Rev 1.4

### Steps to reproduce

None, but I know of some people, which are experiencing the same! 

### Expected behaviour

I would have expected a much quicker migration

### Actual behaviour

Very, very slow migration with hardly any progress

**Comment by C-Otto** [comment]
This migration is very CPU intensive, maybe the Pi isn't quite up for the challenge?

**Comment by positiveblue** [comment]
I think @C-Otto is right here but I understand @Deekee62 concerns because the migration does not take linear time: it took 1h for the first 40% but more than 48h for the next 60%. 

8M entries are quite a lot for a raspberry. There won't be many migrations of this kind, but we will keep in mind to benchmark migration times for different setups so node operators know what to expect.

I will leave this issue open until you give the green light (has been properly migrated) but it looks like it just needs a bit more of time.

**Comment by MEdGalo** [comment]
I'm experiencing the same problem as @Deekee62. I'm running a rpi 4 4gb with umbrel and upgraded from 0.14.2 to 0.15.1 with a channel.db from 13gb. The migration is taking more 24h already and i'm at 86.7%. Htop shows only 1.7gb ram usage and the cpu doesn't seem to be fully stressed. Also in the plebnet groups there are multiple people who are running on a rpi 4 with channel.db that migrated in <1h, but they seemed to have upgraded from 0.15.0.

**Comment by Deekee62** [comment]
Good morning, I already see the migration finishing line after ~64 hrs
`2022-09-06 05:41:05.900 [INF] CHDB: Migration progress: 87.985%, still have: 1002160`

**Comment by Roasbeef** [comment]
We made the migration itself concurrent to be able to take advantage of beefier CPUs. We made it _optional_ as well since we knew that it could take some time for either very large nodes, or nodes with meager hardware. The process is also I/O intensive, so slower disks will make the migration take longer. 

If you were already on 0.15 for some time, then you have less log updates to migrate (the newer ones were already being written to disk).
